### PR TITLE
Feature/models product category module

### PR DESCRIPTION
**Pull request by absarali313**
All models related to Product and Category module

### CODE REVIEW

File: app/Models/Category.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Category extends Model
+{
+    use HasFactory;
+    protected $fillable = ['name','image_url','parent_id'];
```
Comment (MShahnoor) [inline app/Models/Category.php:11]: The fillables should be one at a line as in the following image:
image

File: app/Models/Category.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    public function parent(){
```
Comment (MShahnoor) [inline app/Models/Category.php:13]: There will also be a similar relation for children. A category has a parent and a category also has many children

File: app/Models/Price.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Price extends Model
+{
+    use HasFactory;
+    protected $fillable=[
```
Comment (MShahnoor) [inline app/Models/Price.php:11]: Same indentation issue. Also add a line break before declaration of fillable.

File: app/Models/Product.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Product extends Model
+{
+    use HasFactory;
+
+
```
Comment (MShahnoor) [inline app/Models/Product.php:12]: Be very spacing and indentation. Only 1 line break is needed here. Remove redundant lines.

File: app/Models/Product.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    ];
+
+    public function categories(){
+        return $this->belongsToMany(Category::class,'category_product');
+    }
+    public function relatedProducts(){
+        return $this->belongsToMany(Product::class,'related_products','product_id','related_product_id');
+    }
+
+    public function relatedTo(){
```
Comment (MShahnoor) [inline app/Models/Product.php:27]: RelatedTo isn't clear whether it's a has many relation or belongs to many. Rename it to something that makes more sense, like relatedItems.

File: app/Models/Product.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        'title',
+        'description',
+        'Visibility'
+    ];
+
+    public function categories(){
+        return $this->belongsToMany(Category::class,'category_product');
+    }
+    public function relatedProducts(){
+        return $this->belongsToMany(Product::class,'related_products','product_id','related_product_id');
```
Comment (MShahnoor) [inline app/Models/Product.php:24]: Rename the related_products table to follow the naming convention for middle tables. Check docs for naming convention.

File: app/Models/ProductImage.php [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class ProductImage extends Model
+{
+    use HasFactory;
+    protected $fillable = [
```
Comment (MShahnoor) [inline app/Models/ProductImage.php:11]: Add line break. Also refactor all the files for indentation related issues.

File: app/Models/Product.php [hunk 8]
```diff
@@ -20,18 +18,19 @@ class Product extends Model
     public function categories(){
         return $this->belongsToMany(Category::class,'category_product');
     }
+
     public function relatedProducts(){
```
Comment (MShahnoor) [inline app/Models/Product.php:22]: Why there is a need of two methods for the same purpose? relatedProducts and relatedItems are doing the same thing.

Comment (absarali313) [reply app/Models/Product.php:22]: I think this should be relatedTo and relatedItems like before.
relatedTo will define the inverse relation. Like which products have added this item as related products.

File: app/Models/Category.php (hunk 2 above)
Comment (MShahnoor) [reply app/Models/Category.php:18]: Where is children method?

**Review by MShahnoor — APPROVED** [review]

**Pull request merged**